mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -2650,11 +2713,9 @@ createAlteredByCaptureMap(MapInfoData &mapData,
   for (size_t i = 0; i < mapData.MapClause.size(); ++i) {
     // if it's declare target, skip it, it's handled separately.
     if (!mapData.IsDeclareTarget[i]) {
-      auto mapOp =
-          mlir::dyn_cast_if_present<mlir::omp::MapInfoOp>(mapData.MapClause[i]);
-      mlir::omp::VariableCaptureKind captureKind =
-          mapOp.getMapCaptureType().value_or(
-              mlir::omp::VariableCaptureKind::ByRef);
+      auto mapOp = dyn_cast_if_present<omp::MapInfoOp>(mapData.MapClause[i]);
```

Comment:
Nit: We're using the value right afterwards without checking for null, so at least we should be asserting that it's present and it's the expected type (which we already do in other places where that field is accessed).
```suggestion
      auto mapOp = cast<omp::MapInfoOp>(mapData.MapClause[i]);
```